lib/home_page.dart

```diff
@@ -0,0 +1,314 @@
+                  },
+                  style: TextStyle(color: Colors.white),
+                  decoration: InputDecoration(
+                      icon: Icon(
+                        Icons.search,
+                        color: Colors.white,
+                      ),
+                      hintText: "Search Question",
+                      hintStyle: TextStyle(color: Colors.white)),
+                ),
+          centerTitle: true,
+
+          // Search icon is changed to search bar
+          actions: <Widget>[
+            isSearching
+                ? IconButton(
+                    icon: Icon(Icons.cancel),
+                    onPressed: () {
+                      // Button to drop search bar. Will keep list filtered if category filter is applied
+                      setState(() {
+                        this.isSearching = false;
+                        questionsListFilteredSearch = data; // search list is now the same as the original
+                        filterQuestionsCategory(selectedCategory); // if "All Categories", then it page is completely unfiltered
+                      });
+                    },
+                  )
+                : IconButton(
+                    icon: Icon(Icons.search),
+                    onPressed: () {
+                      setState(() {
+                        this.isSearching = true;
+                      });
+                    },
+                  ),
+
+            // Chat button
+            FlatButton(
+              textColor: Colors.white,
+              onPressed: () {}, // redirect to Chat page
+              child: Icon(Icons.chat_sharp, color: Colors.cyan),
+              shape: CircleBorder(side: BorderSide(color: Colors.transparent)),
+            ),
+          ],
+          backgroundColor: Colors.blueGrey[800],
+        ),
+      ),
+
+      // List of questions and a category filter
+      body: Column(
+        children: [
+          // List of Questions
+          Expanded(
+              // Show loading circle if results are taking time
+              // Show "No results" if input text doesn't match with question title (later will be added to description too)
+              child: _buildList(questionsListFiltered, filterQuestionsCategory, getCategory)),
+
+          // Button "Create New Question" is visible if the user is in Text input mode. This feature is questionable
+          RawMaterialButton(
+            onPressed: () {},
+            fillColor: Colors.blueGrey[800],
+            splashColor: Colors.grey,
+            child: Padding(
+              padding: EdgeInsets.all(10.0),
+              child: SizedBox(
+                width: 200.0,
+                child: Text(
+                  "Create New Question",
+                  maxLines: 1,
+                  style: TextStyle(color: Colors.white),
+                  textAlign: TextAlign.center,
+                ),
+              ),
+            ),
+          ),
+        ],
+      ),
+    );
+  }
+}
```

Comment:
Button for "Create New Question" should not be visible when the user is typing on the search bar. You can extract this for now or comment it out.

Reply:
Ok, I fixed that!
Why would you like to extract "new question" and "chat" buttons though? I think we can keep them.. They just don't have any functionalities yet

Reply:
I guess you could leave it. I just thought it could be a separate issue.